### PR TITLE
Add seaport v1.2 and bulk order signatures

### PR DESCRIPTION
**Pull request by ryanio**
This PR adds Seaport v1.2 and bulk order signatures. Special thanks to @d1ll0n for the great typescript utils ported from the Seaport repo.

TLDR on bulk sigs: The order signature field includes the BulkOrder signature, order index, and proof.

I deduplicated some of the order prep code between `createOrder()` and the new `createBulkOrders()` into a new internal function `_formatOrder()`, to help future maintainability.

There was also both a `package-lock.json` and a `yarn.lock`. Since the readme says to `npm install` rather than use yarn, I kept `package-lock.json` and removed `yarn.lock` and added it to gitignore so it won't accidentally be added in the future.

### CODE REVIEW

File: src/seaport.ts [hunk 1]
```diff
@@ -287,11 +363,10 @@ export class Seaport {
       salt ||
       (domain ? generateRandomSaltWithDomain(domain) : generateRandomSalt());
 
-    const orderParameters: OrderParameters = {
+    const orderComponents: OrderComponents = {
       offerer,
       zone,
-      // TODO: Placeholder
-      zoneHash: formatBytes32String(resolvedCounter.toString()),
+      zoneHash: ethers.constants.HashZero,
```
Comment (ryanio) [inline src/seaport.ts:369]: i'm not sure why the zoneHash was being set to the counter here so i changed it to zero hash

File: .gitignore [hunk 2]
```diff
@@ -4,6 +4,7 @@ coverage
 coverage.json
 typechain
 lib
+yarn.lock
```
Comment (sunwrobert) [inline .gitignore:7]: 👍🏼

File: src/utils/eip712/defaults.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
+/* eslint-disable no-dupe-class-members */
+/* eslint-disable no-unused-vars */
```
Comment (sunwrobert) [inline src/utils/eip712/defaults.ts:2]: these still needed?

**Review by sunwrobert — APPROVED** [review]
lgtm generally

File: src/utils/eip712/defaults.ts (hunk 3 above)
Comment (ryanio) [reply src/utils/eip712/defaults.ts:2]: didnt need no-unused-vars, and i was able to reduce the scope of no-dupe-class-members, pushing a commit for that now 👍

**Pull request merged**